pkg-manifest/read-project-manifest/src/readFile.ts

```diff
@@ -1,9 +1,24 @@
 import gfs from '@pnpm/graceful-fs'
 import { type ProjectManifest } from '@pnpm/types'
+import YAML from 'yaml'
```

Comment:
we use `js-yaml` 
we shouldn't use 2 libraries for parsing yaml.

Reply:
@danielbayley, can you update the suggestion given by @zkochan?

I am desperate to preserve the comments in the `pakcage.yaml`.

Reply:
> we use `js-yaml`
> we shouldn't use 2 libraries for parsing yaml.

@zkochan I agree that 2 is not ideal… I suppose I’m proposing that we replace `js-yaml` with [eemeli.org/`yaml`](https://eemeli.org/yaml), because it retains comments very well, whereas `js-yaml` does not, and instead just discards them. Did you take a look at my working implementation here? https://github.com/pnpm/pnpm/issues/2008#issuecomment-1481770483

> can you update the suggestion given by @zkochan?

@gusbemacbe Unfortunately it isn’t a suggestion that helps with this implementation…

> I am desperate to preserve the comments in the *`package.yaml`.

Same!

Reply:
With a bit more research, I'd also be in favor of a move to [`yaml`](https://eemeli.org/yaml). It looks like `js-yaml` hasn't had a release for a few years.

Switching to `yaml` to avoid 2 libraries for parsing/serializing YAML files will take a bit of work though.

- Is there a performance difference? It'll be interesting to see a performance benchmark on a large `pnpm-lock.yaml` file between `yaml` vs `js-yaml`.
- There's patches to `js-yaml` that we'll need to replicate in `yaml`. https://github.com/nodeca/js-yaml/compare/master...pnpm:js-yaml:features-for-pnpm

@danielbayley Do you happen to have bandwidth for such an investigation? If not, we can spin up a different pull request investigating a switch from `js-yaml` to `yaml` that would merge before this PR.

Reply:
> @danielbayley Do you happen to have bandwidth for such an investigation?

Unfortunately not, currently.

> If not, we can spin up a different pull request investigating a switch from `js-yaml` to `yaml` that would merge before this PR.

@gluxon Fully in favour of the switch… and definitely agree that should be a separate PR, preceding this one. I would be keen to pick up the baton after that to implement comment preservation…

Reply:
Sounds good! Thanks @danielbayley.